### ISSUE TITLE
Crash when pasting a note with a dynamic

### ISSUE THREAD

**Issue by mathesoncalum** [musescore/MuseScore#25676]
### Issue type

Crash or freeze

### Description with steps to reproduce

1. Copy a measure containing a note with a dynamic
2. Paste it
3. Undo
4. Try to paste again to the same spot - MuseScore crashes

It appears we end up in an infinite loop inside `Score::makeGap` - call stack goes:

```
...
NotationActionController::pasteSelection
NotationInteraction::pasteSelection
Score::cmdPaste
Score::pasteStaff
Read410::pasteStaff
Score::makeGap
(infinite loop)
```

### Supporting files, videos and screenshots

https://github.com/user-attachments/assets/0b112757-c8a3-4c9d-87b8-1c57447e2732

### What is the latest version of MuseScore Studio where this issue is present?

4.5 nightly

### Regression

Yes, this works in the latest released version and is broken in the nightly builds

### Operating system

Tested on macOS Sonoma 14.6.1

### Additional context

4.4.4 seems to be fine, crash seems to be present in 4.5 only

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by DmitryArefiev** [comment]
The same on Windows 10 (**4.5 nightly** only)

**Comment by DmitryArefiev** [comment]
Fixed in #25686
